lambda_code/src/fhir_repository.py

```diff
@@ -122,16 +140,22 @@ def create_immunization(self, immunization: dict, patient: dict) -> dict:
                 message="Non-200 response from dynamodb", response=response
             )
 
-    def update_immunization(
-        self, imms_id: str, immunization: dict, patient: dict
-    ) -> dict:
+    def update_immunization(self, imms_id: str, immunization: dict, patient: dict) -> dict:
         attr = RecordAttributes(immunization, patient)
         # "Resource" is a dynamodb reserved word
         update_exp = (
             "SET UpdatedAt = :timestamp, PatientPK = :patient_pk, "
             "PatientSK = :patient_sk, #imms_resource = :imms_resource_val, Patient = :patient"
         )
 
+        queryResponse = _query_identifier(self.table, 'IdentifierGSI', 'IdentifierPK', attr.identifier)
+
+        if queryResponse != None:
```

Comment:
Business logic goes in the service. The repo is just for simple data read and write.